Sources/SwiftGRPCNIO/GRPCClientConnection.swift

```diff
@@ -49,8 +50,23 @@ open class GRPCClientConnection {
       }
 
     return bootstrap.connect(host: host, port: port).flatMap { channel in
-      channel.pipeline.context(handlerType: HTTP2StreamMultiplexer.self).map { context in
-        context.handler as! HTTP2StreamMultiplexer
+      // Check the handshake succeeded and a valid protocol was negotiated via ALPN.
+      let tlsVerified: EventLoopFuture<Void>
+
+      if case .none = tlsMode {
+        tlsVerified = channel.eventLoop.makeSucceededFuture(())
+      } else {
+        tlsVerified = channel.pipeline.context(handlerType: GRPCTLSVerificationHandler.self).map {
+          $0.handler as! GRPCTLSVerificationHandler
+        }.flatMap {
+          $0.verification
```

Comment:
What is this for? Seems like the result gets dropped?

Reply:
We drop the value, but it _could_ contain an `error` which will be propagated back to the caller instead of the connection.

Reply:
But `tlsVerified` is declared as an `ELF<Void>`, not `ELF<Connection>` or something like that? LMK if I'm missing something.

Reply:
We `flatMap` off it a few lines below. If `tlsVerified` contains an error then we won't execute that chain of `map`s, but we will return an `ELF<GRPCClientConnection>` containing the error from `tlsVerified`.

Reply:
Stupid me; I mistook `flatMap` and `map`. Would you mind adding a short comment to this line nevertheless?

Reply:
Easily done! Comment added.